src/warplib/implementations/types/bytes.ts

```diff
@@ -0,0 +1,168 @@
+import endent from 'endent';
+import {
+  forAllWidths,
+  necessary_width_to_store_nlen_bytes,
+  WarplibFunctionInfo,
+} from '../../utils';
+
+export function fixed_bytes_types(): WarplibFunctionInfo {
+  const fixed_bytes_trait = endent`
+    trait FixedBytesTrait<T> {
+      fn atIndex(self: T, index: u8) -> bytes1;
+      fn length(self: T) -> u8;
+      // Comparison
+      fn less_or_equal_comparison(self: T, other: T) -> bool; // a <= b
+      fn less_comparison(self: T, other: T) -> bool; // a < b
+      fn equal_comparison(self: T, other: T) -> bool; // a == b
+      fn distinct_comparison(self: T, other: T) -> bool; // a != b
+      fn greater_or_equal_comparison(self: T, other: T) -> bool; // a >= b
+      fn greater_comparison(self: T, other: T) -> bool; // a > b
+      // Bitwise
+      // TODO: Bitwise is not natively supported for all uN types. We can convert them into the ones that
+      // implement the operation and go back to the original type 
+      fn and_bitwise(self: T, other: T) -> T; // a & b
+      fn or_bitwise(self: T, other: T) -> T; // a | b
+      fn exclusive_or_bitwise(self: T, other: T) -> T; // a ^ b
+      // Bitwise negation is not currently supported
+      // Shift operators are not currently supported
+    }
```

Comment:
Can we store the trait in warplib statically?

Reply:
to do so we would need to create a separate file just for that, then we would have 2 files every time we would want to check something about bytes. Also seems better to have trait definition next to its implementations due that this will not be a commonly extended trait